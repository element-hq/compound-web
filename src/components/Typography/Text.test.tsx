/*
Copyright 2025 New Vector Ltd.
Copyright 2023 The Matrix.org Foundation C.I.C.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

import { Body } from "./Body";

describe("Body", () => {
  it("renders", () => {
    const { container, asFragment } = render(
      <Body>The quick brown fox jumps over the lazy dog.</Body>,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector("p")).not.toBeNull();
  });

  it("lets you override the HTML element", () => {
    const { container } = render(<Body as="span">Hello world.</Body>);
    expect(container.querySelector("p")).toBeNull();
    expect(container.querySelector("span")).not.toBeNull();
  });
});
