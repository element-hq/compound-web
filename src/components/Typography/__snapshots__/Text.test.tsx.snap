// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Body > renders 1`] = `
<DocumentFragment>
  <p
    class="_typography_489030 _font-body-md-regular_489030"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
</DocumentFragment>
`;
